app/components/listing/LocationTimesAccordion.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import { LocationDetails } from "models";
+import classNames from "classnames";
+import styles from "./LocationTimesAccordion.module.scss";
+
+/*
+    Accordion built per a11y guidelines:
+    https://www.w3.org/WAI/ARIA/apg/patterns/accordion/
+*/
+
+const LocationTimesAccordion = ({
+  locationRenderer,
+  locations,
+}: {
+  locations: LocationDetails[];
+  locationRenderer: (loc: LocationDetails) => React.ReactElement;
+}) => {
+  const [activeIndices, setActiveIndices] = useState<number[]>([0]);
+
+  const toggleAccordion = (index: number) => {
+    setActiveIndices((prevActiveIndices) =>
+      prevActiveIndices.includes(index)
+        ? prevActiveIndices.filter((i) => i !== index)
+        : [...prevActiveIndices, index]
+    );
+  };
+
+  return (
+    <div className={styles.locationTimesAccordion}>
+      {locations.map((loc, i) => {
+        const isActive = activeIndices.includes(i);
+        const headerId = `accordion${i}id`;
+        const panelId = `sect${i}`;
+
+        return (
+          <div
+            key={loc.id}
+            className={classNames(
+              styles.accordionItem,
+              isActive && styles.activeItem
+            )}
+          >
+            <h3>
+              <button
+                type="button"
+                aria-expanded={isActive}
+                className={styles.accordionButton}
+                aria-controls={panelId}
+                aria-label={loc.address.address_1}
+                id={headerId}
+                onClick={() => toggleAccordion(i)}
```

Comment:
At some point when I get back lets talk about naming convention for interaction handling on elements. One system I like is using a named callback here like `handleAccordionToggleOnClick` so you have a nice distinction between _handling user intent_ and then the _computational effort_ that follows. IME the handler call space gives you some room for logging (debugging, or monitoring) and easier test doubling to name a couple examples.

Reply:
works for me!